messaging/src/main/scala/uk/ac/wellcome/messaging/worker/steps/MonitoringProcessor.scala

```diff
@@ -1,28 +1,10 @@
 package uk.ac.wellcome.messaging.worker.steps
 
-import java.time.Instant
-
-import uk.ac.wellcome.messaging.worker.logging.Logger
 import uk.ac.wellcome.messaging.worker.models._
-import uk.ac.wellcome.messaging.worker.monitoring._
 
 import scala.concurrent.{ExecutionContext, Future}
 
-trait MonitoringProcessor extends Monitoring with Logger {
-
-  protected def record[Recorded, ProcessMonitoringClient <: MonitoringClient](
-    startTime: Instant,
-    result: Result[Recorded]
-  )(implicit monitoringClient: ProcessMonitoringClient,
-    ec: ExecutionContext): Future[Result[Unit]] = {
-
-    val monitoring = for {
-      _: Unit <- log(result)
-      _: Unit <- metric(result, startTime)
-    } yield Successful[Unit]()
-
-    monitoring recover {
-      case e => MonitoringProcessorFailure[Unit](e)
-    }
-  }
+trait MonitoringProcessor[Message, Context]{
+  def recordStart(message: Message)(implicit ec: ExecutionContext): Future[Context]
+  def recordEnd[Recorded](message: Message, context: Context, result: Result[Recorded])(implicit ec: ExecutionContext): Future[Result[Unit]]
```

Comment:
Do we run `scalafmt` on this repo? Line length not working out for me :see_no_evil:

Reply:
Ah I though Travis would that but I guess not..